buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -402,7 +402,9 @@ void reallocate(PooledByteBuf<T> buf, int newCapacity, boolean freeOldMemory) {
             buf.trimIndicesToCapacity(newCapacity);
             bytesToCopy = newCapacity;
         }
-        memoryCopy(oldMemory, oldOffset, buf.memory, buf.offset, bytesToCopy);
+        if (bytesToCopy > 0) {
+            oldNioBuffer = memoryCopy(oldMemory, oldNioBuffer, oldOffset, buf, bytesToCopy);
```

Comment:
hmm is this really correct ? This may now be duplicate that is passed to free(...)

Reply:
@normanmaurer the `oldNioBuffer` passed there is already a duplicate (comes from `buf.tmpNioBuf`) and isn't free'd itself, rather kept in the cache for later re-use by other `ByteBuf`s that share the same "root" nio buffer...